docs/syntax-attribute-functions-native.md

```diff
@@ -0,0 +1,54 @@
+# Native Attribute for Functions
+
+## Summary
+
+This RFC proposes a function-level `@native` attribute to request native compilation for individual functions, independent of the script-level `--!native` comment directive.
+
+## Motivation
+
+Luau's native compiler currently compiles whole scripts annotated with `--!native` comment directive. However, this provides very coarse-grained control. Since all functions in the script may not benefit from native compilation, developers might be forced to move unrelated functions together to natively compiled scripts. In this RFC, we propose a function-level `@native` attribute to facilitate developers to pick and choose individual functions for native compilation.
+
+## Design
+
+Syntactically, the `@native` attribute takes no parameters. It can be used on both top-level and inner functions. It applies recursively to all the functions defined within the lexical scope of the attributed function because the "inner" functions logically constitute the implementation of the attributed function. Hence, attributing "inner" functions as `@native` will not have any effect if an _ancestor_ function already has a `@native` attribute. 
+
+In the following example, both `parent` and `child` will be natively compiled. 
+
+```lua
+@native
```

Comment:
Maybe this is actually expected, but for me personally, this feels like an ick. I don't know if I like the idea that `@native` is applied recursively.

Reply:
Reversed this behavior.